generator.cpp

```diff
@@ -13,17 +13,17 @@ std::unordered_set<uint64_t> visited;
  * Explore and print all possible position under a given depth.
  * symetric positions are printed only once.
  */
-void explore(const Position &P, char* pos_str, const int depth) {
+void explore(const Position &P, char* const pos_str, const int depth) {
   uint64_t key = P.key3();
   if(visited.count(key)) return;  // already explored position
   visited.insert(key);            // flag new position as visited
```

Comment:
You should really consider using the error thrown here: https://en.cppreference.com/w/cpp/container/unordered_set/insert if the iterator returned form `.insert` is something other than `.end`. This will save a lookup in the table.

Reply:
Good point, I took your suggestion into account in my last commit